File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        ind = 0
        b = True
        for i in range(len(word)):
            if word[i]==ch:
                ind = i
                b = False
                break;
        if b:
            return word
        res = word[0:i+1]
        res = res[::-1]
        res += word[i+1::]
        return res

File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.cpp
class Solution {
public:
    string reversePrefix(string word, char ch) {
        int ind=0;
        bool b = true;
        string res;
        for(int i=0 ; i<word.size() ; i++){
            res+= word[i];
            if(word[i]==ch){
                ind = i;
                b = false;
                break;
            }
        }
        if(b) return word;
        reverse(res.begin(), res.end());
        for(int i=ind+1 ; i<word.size() ; i++){
            res+= word[i];
        }
        return res;
    }
};